react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: 'email' };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const value = event.target.value;
+    this.setState({ email: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log('state:', this.state);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label htmlFor='email-input'>
+          Email
+        </label>
+        <input type="email" id="email-input" value={this.state.email} onChange={this.handleChange}>
```

Comment:
```jsx
value={this.state.email}
```

the `value` prop needs to be passed the right state property. yours is named `value`.

```jsx
value={this.state.value}
```